utils/strings.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2017 The kubecfg authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import (
+	"strings"
+)
+
+// HasSpecialCharacter takes a string and returns true if the string contains
+// any special characters.
+func HasSpecialCharacter(s string) bool {
+	f := func(r rune) bool {
+		return r < 'A' || r > 'z'
```

Comment:
So, formally, Jsonnet can't lex unicode identifiers. So this logic is probably what we want. But, I think `HasSpecialCharacters` is a little opaque. I would call it `IsASCIIIdentifier` and invert the logic. Because that's what we want: if `!IsASCIIIdentifier` then quote.